runtime-light/tl/tl-types.h

```diff
@@ -22,4 +24,20 @@ struct K2JobWorkerResponse final {
   void store(TLBuffer &tlb) const noexcept;
 };
 
+// ===== CRYPTO =====
+
+enum CertInfoItem : uint32_t {
```

Comment:
Should it really be declared in the header file? Is it necessary that clients know these magics?

Reply:
It's actually one type, but with three magics
I think it's better than have three separate constexpr vars

Reply:
No problem with having them as enumeration. I just think that they should not be a part of public interface of tl namespace

Reply:
What whould you suggest?

Reply:
Move it to cpp file, for example

Reply:
`K2_JOB_WORKER_RESPONSE_MAGIC` is used by clients outside of `tl` namespace.

Reply:
Where? I see it only in` tl-types.cpp`

Reply:
Yep, you're right. Anyway, these magics should only be used to "detect" that we got a specific tl-type or tl-function. I think we should hide all the details about the implementation. `CertInfoItem` looks like implementation detail.

Reply:
Your logic is following: magics for functions and request/response types should be public, magic of everything else should be hidden in `.cpp` file, am I right?

Reply:
I think that every high-level entity should have a publicly available magic, while all the implementation details must be hidden

Reply:
Who defines whether it "high-level entity" or "implementation details"?
What is specific criteria?

In my opinion, all TL stuff is implementation details regarding other runtime code

Reply:
We do, of course :) For example, we can use following rule: everything that is only used inside some TL-type of -function should be hidden.

Reply:
OK, according to this rule all the magics (for now, at least) should be hidden?
Because all magics are used only inside some TL-combinator entity -- for now, all of them has `fetch` and/or `store` methods which are the single use cases

Reply:
Decided to move `CertInfoItem` into `.cpp` and later all magics defined in `.cpp`.